src/Datadog.Trace.ClrProfiler.Native/pal.h

```diff
@@ -0,0 +1,58 @@
+#ifndef DD_CLR_PROFILER_PAL_H_
+#define DD_CLR_PROFILER_PAL_H_
+
+#ifdef _WIN32
+
+#include <process.h>
+#include <filesystem>
+#include "windows.h"
+
+#else
+
+#include <unistd.h>
+#include <fstream>
+
+#endif
+
+#include "string.h"
+
+namespace trace {
+
+inline WSTRING DatadogLogFilePath() {
+#ifdef _WIN32
+  std::string programdata(getenv("PROGRAMDATA"));
+  if (programdata.empty()) {
+    programdata = "C:\\ProgramData";
+  }
+  return ToWSTRING(programdata + "\\Datadog\\logs\\dotnet-profiler.log");
```

Comment:
Let's use `C:\ProgramData\Datadog .NET Tracer\logs` to avoid conflicts with the Agent. The Agent sets stronger permissions on its folder than it should, and apps hosted on IIS can't write there by default.

Reply:
@dd-caleb: I should've said `C:\ProgramData\Datadog .NET Tracer\logs\dotnet-profiler.log`. I don't think we want to drop the filename.